srcpkgs/ldb/template

```diff
@@ -1,29 +1,76 @@
 # Template file for 'ldb'
 pkgname=ldb
-version=1.5.1
+version=2.2.0
 revision=1
-build_style=gnu-configure
-configure_args="--with-modulesdir=/usr/lib/ldb/modules"
-hostmakedepends="python pkg-config docbook-xsl libxslt"
-makedepends="tdb-devel tevent-devel popt-devel talloc-devel lmdb-devel"
+build_style=waf3
+build_helper="qemu"
+configure_script="buildtools/bin/waf"
+configure_args="--sysconfdir=/etc --localstatedir=/var
+ --disable-rpath --disable-rpath-install --without-gettext
+ --builtin-libraries=replace --bundled-libraries=NONE
+ --with-modulesdir=/usr/lib/ldb/modules"
+hostmakedepends="pkg-config docbook2x tdb-python3 tevent-python3"
+makedepends="python3-devel tdb-devel tevent-devel talloc-devel
+ popt-devel lmdb-devel libxslt gettext-devel cmocka-devel"
 checkdepends="cmocka-devel"
 short_desc="LDAP-like database from samba"
 maintainer="Yuusha Spacewolf <xyuusha@paranoici.org>"
 license="LGPL-3.0-or-later"
 homepage="https://www.samba.org/ldb/"
 distfiles="https://www.samba.org/ftp/pub/ldb/ldb-${version}.tar.gz"
-checksum=232e54b87c53210a6861424ad411b04dbfa0f2ca4bc463aefea188158e0cc963
+checksum=134bb51769709af59f30bf468e454d1377a8096acd4e80dcb42fd264f558bd5f
 
-nocross="see srcpkgs/tdb (cross-32bit.answers)."
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --cross-compile --hostcc=cc
+	 --cross-execute=qemu-${XBPS_TARGET_QEMU_MACHINE}-static"
+fi
 
 # workaround for cmocka's broken uintptr_t definition on musl
 if [ "$XBPS_TARGET_WORDSIZE" = "64" -a "$XBPS_TARGET_LIBC" = "musl" ]; then
 	CFLAGS+=" -D__WORDSIZE=64"
 fi
 
+post_extract() {
+	# wrap target python3.9-config to find the right python flags
+	if [ "$CROSS_BUILD" ]; then
+		local _pyconfig="${XBPS_WRAPPERDIR}/python${py3_ver}-config"
+		cat> "${_pyconfig}" <<-EOF
+		#!/bin/sh
+		exec "${XBPS_CROSS_BASE}/usr/bin/python${py3_ver}-config" "\$@"
+		EOF
+		chmod 755 "${_pyconfig}"
+	fi
```

Comment:
Could we make this default?

Reply:
I was thinking about that. It would be nice, but I don't know about the ramifications for other cross-built applications that rely on the current, more simplistic wrapper.

Reply:
The system's `python3-config` requires `which`. Which means we need to either add `which` to `hostmakedepends` for all the things that need `python3-config` or make `python3-devel` depends on `which`.

Reply:
Fixed in the new `samba` build helper, will push soon.